### ISSUE TITLE
Git check hangs indefinitely

### ISSUE THREAD

**Issue by hexedpackets** [concourse/git-resource#81]
I'm running Concourse 2.1.0 and occasionally see git resources lock up. In happens across all pipelines without any error message. Attempting a `check-resource` via fly also hangs with no output. This seems similar to #68, although I'm not using btrfs.

I've tried aborting and restarting jobs without any luck. I've also left a job running overnight but came back to it still waiting on the git resource. Deleting and recreating the pipeline does allow it to immediately succeed.

**Comment by vito** [comment]
I'm afraid there's not much we can do about this without a way to reproduce it. It could be anything; network issues talking to the git repo, a deadlock (in postgres, `git`, something unpredicted), etc. etc. Is this still happening? If so, can you describe your deployment?

**Comment by kidbrax** [comment]
I am seeing this same behavior.  check-resource just sits there forever.  As mentioned, if I delete the pipeline completely and re-add it, then check-resource seems to work.  And while you may not be able to reproduce this, the task should time out at some point anyway, right?  No matter what happens, at some point it should at least return to the prompt and output something.

**Comment by kidbrax** [comment]
I also noticed that I can delete/re-add the pipeline and then check-resource works fine.  But as soon as i run a build from the web interface, it will stick at checking git and check-resource will then fail from the command line.

**Comment by alejandrocacho** [comment]
Same problem here. It´s been more than 2 years since the original comment, come on guys, you cannot reproduce something similar? Can´t you add a timeout as @kidbrax suggested?

**Comment by vito** [comment]
@alejandrocacho You're surprised that we haven't looked at a *closed issue* for two years?

If this is affecting you, please consider opening a new issue or - better yet - a pull request. Open-source projects aren't a bunch of people slaving away to solve your problems for free. Consider contributing yourself.
